src/lib/plugins/log.js

```diff
@@ -1,20 +1,39 @@
 const fs = require('fs')
-const timeStarted = Math.floor(new Date() / 1000).toString()
+const timeStarted = Math.floor(Date.now() / 1000).toString()
 const path = require('path')
-const mkdirp = require('mkdirp')
 const moment = require('moment')
 const colors = require('colors')
 
-const readline = require('readline')
-const rl = readline.createInterface({
-  input: process.stdin,
-  output: process.stdout
-})
+const isInNode = typeof process !== 'undefined' && !process.browser && process.platform !== 'browser'
 
-rl.setPrompt('> ')
-rl.prompt(true)
+const _servers = []
+
+/** @type {typeof import("readline") | undefined} */
+let readline
+/** @type {import("readline").Interface | undefined} */
+let rl
+if (isInNode) {
+  import(/* webpackIgnore: true */ 'exit-hook').then((hook) => {
+    hook.default(() => {
+      for (const serv of _servers) {
```

Comment:
In the context of this refactor also fixed an important bug: worlds are not saved when server crashed or stopped by ctrl+c

Reply:
I understand the intent and I agree the feature is nice. However that couple the lib and the app parts of flying-squid
What about instead define this as some ServerManager class and use it in app.js ?

Reply:
What about exit-hook: this is totally okay to have it as part of the library and why not have it so "better behavior is enabled by default"?

Reply:
Hm, as I understand there is also no way to disable readline, right?

Reply:
"exit-hook: this is totally okay to have it as part of the library"

no that's exactly the part I don't like.

I think it's good if the library exposes a function that makes this possible, and then it's called by default in https://github.com/PrismarineJS/flying-squid/blob/master/app.js

having the library actually call it forces the user into a specific application setup

Reply:
> no that's exactly the part I don't like.

Alright, I don't mind making it an opt-in behavior, however, then I think it would make sense to make creating readline interface optional as well. What do you think?
Otherwise I might be missing use-cases of this library (sorry)

> I think it's good if the library exposes a function that makes this possible

Will extract, doc later today or tomorrow, not a problem

Reply:
> I think it's good if the library exposes a function that makes this possible, and then it's called by default 

Maybe this should be done as option? like `registerExitHook: true` ?